flask_wiki/api.py

```diff
@@ -399,24 +389,37 @@ def current_language(self):
     def languages(self):
         return current_app.config.get('WIKI_LANGUAGES')
 
-    def search(self, term, ignore_case=True, attrs=['title', 'tags', 'body']):
+    def search(self, term, ignore_case=True, attrs=None):
+        if attrs is None:
+            attrs = ['title', 'tags', 'body']
         pages = self.index()
 
+        for page in pages:
+            page["score"] = 0
+
+        # When searching for "*", return ALL pages
         if term == "*":
             return pages
 
-        current_language_pages = [p for p in pages if p.language == self.current_language]
+        current_language_pages = [
+            p for p in pages if p.language == self.current_language]
+
+        # If no query term, return all current language pages
         if not term:
             return current_language_pages
 
-        regex = re.compile(term, re.IGNORECASE if ignore_case else 0)
+        regex = re.compile(
+            re.escape(term), re.IGNORECASE if ignore_case else 0)
+
         matched = []
         for page in current_language_pages:
             for attr in attrs:
-                if regex.search(getattr(page, attr)):
-                    matched.append(page)
-                    break
-        return matched
+                if found := re.findall(regex, getattr(page, attr)):
+                    page["score"] += len(found)
```

Comment:
`page['score'] = page.setdefault('score', 0) + len(found)`

Reply:
I just tried this and it prompts an error because `page` is a `Page` object and not a dictionnary.